### DISCUSSION TITLE
One-to-One Relationship Using Parent Table Primary Key as Primary and Foreign Key

### DISCUSSION THREAD

**Discussion by iampopal** [serverpod/serverpod#2886] [Q&A]
I've been working with one-to-one relationships in Serverpod, and I noticed that the system auto-generates a primary key for the child table, even in cases where a one-to-one relationship exists. My understanding is that in a one-to-one relationship, it is often preferable (or possible) for the child table to use the parent table’s primary key as both the primary and foreign key, ensuring a strict one-to-one correspondence between the two tables.

My question is:

Is it possible in Serverpod to configure a one-to-one relationship where the child table uses the parent table's primary key as both its primary key and foreign key, instead of generating a separate primary key for the child table?
If not currently supported, are there any plans to introduce this capability, or is there an alternative best practice within the Serverpod framework?

Looking forward to your guidance and insight!

**Comment by iampopal** [comment]
@SandPod please have a look here.

**Comment by SandPod** [comment]
Hey @iampopal!

I'm not sure I understand your question.

If the parent and child tables share the same primary key, what would be the reason for separating the data in different tables?

You can enforce a strict one-to-one relationship by utilizing indexes as shown in the example here: https://docs.serverpod.dev/concepts/database/relations/one-to-one#with-an-object

**Comment by iampopal** [comment]
It’s a one to one relationship

For example between table of customers and tax_no

Each customer ha one tax_no
And each tax_no has one customer

And there are 80% of customers not having tax_no. So to minimize null’s
value we have created a table

customer_tax_no:
fields:
      customer_id
      tax_no

Now if a person save tax no. It will get saved in customer_tax_no table as
a unique entry.

But in serverpod it always create a primaryKey of id and in our case we do
not need a unique id because each customer_id is a unique id in
customer_tax table.

**Reply by SandPod** [reply]
OK, this is not supported in Serverpod.

Each row in the database gets a unique ID for that table.

**Reply by iampopal** [reply]
Can we create an issue for this because adding this will be very helpful

**Reply by SandPod** [reply]
Of course! Please create an issue for it where you provide an example of how you would like it to work and what you would use it for.

You can refer to this thread from the issue.

I don't think it will be prioritized anytime soon because the functionality can be supported through the current system (unique ids for each table).
